src/Cli/dotnet/commands/dotnet-info/InfoCommand.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using Microsoft.DotNet.Cli;
+using System.Text.Json;  
+using Microsoft.DotNet.Cli.Utils;
+
+using LocalizableStrings = Microsoft.DotNet.Cli.Utils.LocalizableStrings;
+using RuntimeEnvironment = Microsoft.DotNet.Cli.Utils.RuntimeEnvironment;
+
+namespace Microsoft.DotNet.Tools.Info
+{
+    public class InfoCommand
+    {
+        private readonly ParseResult _parseResult;
+
+        public InfoCommand(ParseResult parseResult)
+        {
+            _parseResult = parseResult;
+        }
+
+        public static int Run(ParseResult result)
+        {
+            result.HandleDebugSwitch();
+            var format = result.GetValue(InfoCommandParser.FormatOption);
+            if (format != InfoCommandParser.FormatOptions.json) {
+                PrintInfo();
+            } else  {
+                // To be implemented
+                PrintJsonInfo();
+            }
+            return 0;
+        }
+
+        public static void PrintVersion()
+        {
+            Reporter.Output.WriteLine(Product.Version);
+        }
+
+        public static void PrintInfo()
+        {
+            DotnetVersionFile versionFile = DotnetFiles.VersionFileObject;
+            var commitSha = versionFile.CommitSha ?? "N/A";
+            Reporter.Output.WriteLine($"{LocalizableStrings.DotNetSdkInfoLabel}");
+            Reporter.Output.WriteLine($" Version:           {Product.Version}");
+            Reporter.Output.WriteLine($" Commit:            {commitSha}");
+            Reporter.Output.WriteLine($" Workload version:  {WorkloadCommandParser.GetWorkloadsVersion()}");
+            Reporter.Output.WriteLine();
+            Reporter.Output.WriteLine($"{LocalizableStrings.DotNetRuntimeInfoLabel}");
+            Reporter.Output.WriteLine($" OS Name:     {RuntimeEnvironment.OperatingSystem}");
+            Reporter.Output.WriteLine($" OS Version:  {RuntimeEnvironment.OperatingSystemVersion}");
+            Reporter.Output.WriteLine($" OS Platform: {RuntimeEnvironment.OperatingSystemPlatform}");
+            Reporter.Output.WriteLine($" RID:         {GetDisplayRid(versionFile)}");
+            Reporter.Output.WriteLine($" Base Path:   {AppContext.BaseDirectory}");
+            PrintWorkloadsInfo();
```

Comment:
Thank you for the feedback and apologies for the late response! I was exploring how to integrate the information into the new info command. I realized that these information are hard coded for printing in [some C++ files](https://github.com/dotnet/runtime/blob/main/src/native/corehost/fxr/command_line.cpp#L283) in `dotnet/runtime`.

Though we have ways to acquire information about SDKs installed and runtimes installed from the C++ files: [`Interop.cs`](https://github.com/dotnet/sdk/blob/main/src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs#L131), as currently, I didn't find ways to acquire Host and Other architectures file information within `dotnet/sdk`. Therefore, A new standalone info command may need to make modifications to the `dotnet/runtime` project in this [file](https://github.com/dotnet/runtime/blob/f0a6dbdf654b07cedc05217824500ae653409880/src/native/corehost/fxr/hostfxr.cpp#L356), by adding get_environmental_variable, get_host_info, get_global_json functions.

If you happen to know any other way to get host information without making modifications to `dotnet/runtime`, or any existing methods within `dotnet/sdk` to get those information, it would be really helpful. Making changes to `dotnet/runtime`, which involves complex and low-level C++ code, is a significant undertaking and difficult to debug with.

Reply:
Options:
1. Expose the missing pieces of information from corehost so SDK can utilize it via interop for this new command.
2. We are currently using RapidJSON in corehost to parse runtime json files. We can use it to emit info in JSON format when -f json arg (or any of its variants; --format json, --format=json) is set in a separate function like `print_muxer_info_json`, which will inject the host-specific parts (not very clean solution but doable..)
3. Without changing the corehost; spawn a process with `unset unset DOTNET_ROOT;  dotnet --info` command, capture and parse result, convert to JSON and print. (worse than #2 because process spawning is expensive and not provided by supported platforms)

I think the first option is preferred, but I will defer to @elinor-fung and @vitek-karas. Lets hear their thoughts. :)

Reply:
^^

Reply:
I would prefer to do everything in managed code - we've discussed this in the past and the consensus was that the current design where part of the output is produced by the managed code and part of it is from native code is not what we want. Some of the downsides:
* It produces partially localized output - the host parts are not localized (ever), while the SDK parts are.
* It's complex to maintain (and confusing to work on).

As to what mechanism to use to expose the necessary information to managed code:
* Introduce a new `hostfxr` API - or rather modify the existing. Effectively modify this struct to include the necessary info: https://github.com/dotnet/runtime/blob/c28bec4d3d63849c9e60dee1e7174b9a180a7e55/src/native/corehost/hostfxr.h#L311-L323.
* Expose this information as runtime properties. This would have to use the new "on demand" runtime properties which are not yet wired to `AppContext.GetData`, so that would have to be added as well.

Personally - this feels more aligned with the `hostfxr_dotnet_environment_info`. But I don't feel strongly either way.